.circleci/config.yml

```diff
@@ -129,11 +129,31 @@ jobs:
           name: Terraform validate
           command: terraform validate
           working_directory: terraform/
+  build-docker:
+    machine:
+      image: ubuntu-1604:201903-01
+    steps:
+      - checkout
+      - run:
+          name: Check if the docker build job should run or not
+          command: .circleci/should_build_docker.sh
+      - run:
+          name: Get the list of updated docker files
+          command: echo 'export DOCKER_FILES=$(find $(git diff-tree --no-commit-id --name-only -r HEAD) -name "*.Dockerfile")' >> $BASH_ENV
+      - run:
+          name: Build each of the updated docker files
+          command: |
+            for docker_file in $DOCKER_FILES; do
+              file_no_ext=$(basename $docker_file)
+              module=${file_no_ext%.Dockerfile}
+              docker build -f $docker_file --tag commit_verify_${CIRCLE_BUILD_NUM}_$module .
+            done
 
 workflows:
   commit-workflow:
     jobs:
       - build
+      - build-docker
```

Comment:
How long will this step take? if it takes longer than roughly ~14 min then I don't think we can do this as it would dramatically slow down lands and feedback loop from CI

Reply:
Ok I think i understand a little bit more, you only want to run this step when the dockerfiles change which *might* be ok

Reply:
This step takes ~1hr.  It only triggers on a commit that contains change in Dockerfile.  It then builds only the changed Dockerfiles.